assertj-error-prone/src/main/java/com/palantir/assertj/errorprone/AssertjRefactoring.java

```diff
@@ -0,0 +1,572 @@
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchSynchronized(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchThrow(ThrowTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchThrow(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchTry(TryTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchTry(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchTypeCast(TypeCastTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchTypeCast(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchTypeParameter(TypeParameterTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchTypeParameter(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchUnary(UnaryTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchUnary(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchUnionType(UnionTypeTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchUnionType(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchVariable(VariableTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchVariable(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchWhileLoop(WhileLoopTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchWhileLoop(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchWildcard(WildcardTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchWildcard(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    private void describe(Optional<AssertjCheckerResult> result, Tree tree, VisitorState state) {
+        if (result.isPresent()) {
+            AssertjCheckerResult value = result.get();
+            state.reportMatch(buildDescription(tree).setMessage(value.description()).addFix(value.fix()).build());
```

Comment:
nit: may just be the formatter but would be nice on separate lines

Reply:
interesting - so the `reportMatch` thing has the desired effect even though all methods return Description.NO_MATCH?

Reply:
Isn't this what the formatter was supposed to solve ;-)
I wrote it with each invocation on a newline and agree that's more readable. Perhaps @dansanduleac can do some magic when he's back. I don't much mind either way though.

Reply:
> so the reportMatch thing has the desired effect even though all methods return Description.NO_MATCH?

That's correct. I did a similar thing in CatchSpecificity and BracesRequired in baseline, based on FieldCanBeFinal upstream!

Reply:
yeah this repo is locked into the formatter, so even if you tried to put newlines here it would put em back.